src/topp/DecoyDatabase.cpp

```diff
@@ -165,173 +186,234 @@ class TOPPDecoyDatabase :
     f.writeStart(out);
     FASTAFile::FASTAEntry entry;
 
-    // Configure Enzymatic digestion
-    // TODO: allow user-specified regex
+    // Create a ProteaseDigestion object for peptide digestion
     ProteaseDigestion digestion;
     String enzyme = getStringOption_("enzyme").trim();
-    if ((input_type == SeqType::protein) && !enzyme.empty())
-    {
-      digestion.setEnzyme(enzyme);
-    }
-    // check if decoy_string is common decoy string (e.g. decoy, rev, ...)
-    String decoy_string_lower = decoy_string;
-    decoy_string_lower.toLower();
-    bool is_common = false;
-    for (const auto& a : DecoyHelper::affixes)
+    if ((input_type == SeqType::protein) && ! enzyme.empty()) { digestion.setEnzyme(enzyme); }
+    
+    
+ // create neighbor peptides for the relevant peptides given in the FASTA from '-in'
+    String neighbor_file = getStringOption_("Neighbor_Search:neighbor_in");
+ // Check if the neighbor file option is provided
+    if (! neighbor_file.empty())
     {
-      if ((decoy_string_lower.hasPrefix(a) && decoy_string_position_prefix) || (decoy_string_lower.hasSuffix(a) && !decoy_string_position_prefix))
+      String out_neighbor = getStringOption_("out_neighbor");
```

Comment:
add a check, that the user selected 'protein' as input type. For neighbors and DNA are incompatible.
```
if (input_type != SeqType::protein)
{
  OPENMS_LOG_ERROR << "Parameter settings are invalid. When requesting neighbor peptides, the input type must be 'protein', not 'DNA'.\n";
  return INCOMPATIBLE_INPUT_DATA;
}
```

Reply:
I change from DNA to RNA because the type only selected RNA and peptides